src/main/java/pl/numerki/backend/LagrangeInterpolation.java

```diff
@@ -0,0 +1,43 @@
+package pl.numerki.backend;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class LagrangeInterpolation {
+    public static Function interpolate(double[] x, double[] y) {
+        double[] factors = new double[x.length];
+        for (int i = 0; i < x.length; i++) {
+            double xValues = 1;
+            for (int j = 0; j < x.length; j++) {
+                if (i != j) {
+                    xValues *= x[i] - x[j];
+                }
+            }
+            factors[i] = y[i] / xValues;
+        }
+        double[] interpolatedFunctionFactors = new double[x.length];
+        for (int i = 0; i < x.length; i++) {
+            double[] polynomialFactors = {1};
+            for (int j = 0; j < x.length; j++) {
+                if (i != j) {
+                    double[] multiplyFactors = {-x[j], 1};
+                    polynomialFactors = functionMultiPly(polynomialFactors, multiplyFactors);
+                }
+            }
+            for (int j = 0; j < polynomialFactors.length; j++) {
+                interpolatedFunctionFactors[j] += polynomialFactors[j] * factors[i];
+            }
+        }
+        return Functions.polynomial(interpolatedFunctionFactors);
```

Comment:
I think, this can be simplified to:
 ```suggestion
        double[] interpolatedFunctionFactors = new double[x.length];
        for (int i = 0; i < x.length; i++) {
            double xValues = 1;
            double[] polynomialFactors = {1};
            for (int j = 0; j < x.length; j++) {
                if (i != j) {
                    xValues *= x[i] - x[j];
                    double[] multiplyFactors = {-x[j], 1};
                    polynomialFactors = functionMultiPly(polynomialFactors, multiplyFactors);
                }
            }
            for (int j = 0; j < polynomialFactors.length; j++) {
                interpolatedFunctionFactors[j] += y[i] * polynomialFactors[j] * xValues;
            }
        }
        return Functions.polynomial(interpolatedFunctionFactors);
```